docs/application/dotnet/guides/nui/widget.md

```diff
@@ -0,0 +1,257 @@
+      ```
+
+  - The `OnResume()` callback is triggered when the widget instance is resumed.
+
+      Take the necessary actions when the widget instance becomes visible.
+
+      ```csharp
+      protected override void OnResume()
+      ```
+
+  - The `OnResize()` callback is triggered when the widget instance is resized.
+
+      Take the necessary actions to accommodate the new size.
+      ```csharp
+      protected override void OnResize(Window window)
+      ```
+
+  - The `OnUpdate()` callback is triggered when a widget update event is received. 
+
+      Take the necessary actions for the widget update. If the isForce parameter is true, the widget can be updated even in the pause state.
+      ```csharp
+      protected override void OnUpdate(string contentInfo, int force)
+      ```
+
+6. Drawing the Widget UI in OnCreate()
+
+    The widget UI is drawn in the `OnCreate()` callback of your widget class:
+
+    ```csharp
+    protected override void OnCreate(string contentInfo, Window window)
+    {
+        View rootView = new View();
+        rootView.BackgroundColor = Color.White;
+        rootView.Size2D = window.Size;
+        rootView.PivotPoint = PivotPoint.Center;
+        window.GetDefaultLayer().Add(rootView);
+
+        TextLabel sampleLabel = new TextLabel("Hello World!");
+        sampleLabel.FontFamily = "SamsungOneUI 500";
+        sampleLabel.PointSize = 71;
+        sampleLabel.TextColor = Color.Black;
+        sampleLabel.SizeWidth = 200;
+        sampleLabel.PivotPoint = PivotPoint.Center;
+
+        rootView.Add(sampleLabel);
+    }
+    ```
+
+## Create a NUI Widget View
+
+If necessary, you can create a Widget Viewer that shows UI drawn by Widget
+
+The steps to create a nui widget view:<br>
+1. Add privileges for viewer to tizen-manifest.xml
+
+```XML
+  <privileges>
+    <privilege>http://tizen.org/privilege/widget.viewer</privilege>
+    <privilege>http://tizen.org/privilege/appmanager.launch</privilege>
+  </privileges>
+```
+2. Create widget manager and request add widget to widget manager
+
+- Using the `NUIWidgetApplication(Type type)` constructor, add a widget using widget app ID.
+```csharp
+protected override void OnCreate()
+{
+  WidgetViewManager widgetViewManager = new WidgetViewManager(this, this.ApplicationInfo.ApplicationId);
+
+  WidgetView myWidgetView = widgetViewManager.AddWidget("org.tizen.example.Widget", encodedBundle, width, height, 0);
+  Window.Instance.GetDefaultLayer().Add(myWidgetView);
+```
+
+- Using the `NUIWidgetApplication(Dictionary< Type, string > typeInfo) ` constructor, add a widget using class and app ID
+```csharp
+protected override void OnCreate()
+{
+  WidgetViewManager widgetViewManager = new WidgetViewManager(this, this.ApplicationInfo.ApplicationId);
+
+  WidgetView myWidgetVie1 = widgetViewManager.AddWidget("second@org.tizen.example.Widget", encodedBundle, width, height, 0);
+  Window.Instance.GetDefaultLayer().Add(myWidgetView);
+```
```

Comment:
Insert 4 spaces in front of this paragraph.

```suggestion
2. Create a widget manager and request for adding a widget to the widget manager.

    You can create a widget manager using two different constructors:
    
    - The `NUIWidgetApplication(Type type)` constructor for adding a widget using the widget app ID.
       ```csharp
        protected override void OnCreate()
        {
           WidgetViewManager widgetViewManager = new WidgetViewManager(this, this.ApplicationInfo.ApplicationId);
          WidgetView myWidgetView = widgetViewManager.AddWidget("org.tizen.example.Widget", encodedBundle, width, height, 0);
          Window.Instance.GetDefaultLayer().Add(myWidgetView);
          ...  
          }
        ```
    - The `NUIWidgetApplication(Dictionary< Type, string > typeInfo) ` constructor for adding a widget using class and app ID.

        ```csharp
        protected override void OnCreate()
        {
          WidgetViewManager widgetViewManager = new WidgetViewManager(this, this.ApplicationInfo.ApplicationId);
          WidgetView myWidgetVie1 = widgetViewManager.AddWidget("second@org.tizen.example.Widget",         encodedBundle, width, height, 0);
          Window.Instance.GetDefaultLayer().Add(myWidgetView);
          ...
        ```
```
--- 

By the way, the code snippet is the same except values, myWidgetView and myWidgetVie1.

As I understand, the following paragraph is better:
```suggestion

2. Create a widget manager and request for adding a widget to the widget manager.

    You can create a widget manager in `onCreate()` method:
     ```csharp
     protected override void OnCreate()
     {
           WidgetViewManager widgetViewManager = new WidgetViewManager(this, this.ApplicationInfo.ApplicationId);
          WidgetView myWidgetView = widgetViewManager.AddWidget("org.tizen.example.Widget", encodedBundle, width, height, 0);
          Window.Instance.GetDefaultLayer().Add(myWidgetView);
          ...  
      }